tests/integration/cairo1/hint_locals/start_warp/start_warp_test.cairo

```diff
@@ -0,0 +1,98 @@
+    let target = deploy_contract('simple', ArrayTrait::new()).unwrap();
+    assert(target != 0, 'target != 0');
+
+    let proxy = deploy_contract('proxy', ArrayTrait::new()).unwrap();
+    assert(proxy != 0, 'proxy != 0');
+
+    start_warp(100, target);
+
+    let mut calldata = ArrayTrait::new();
+    calldata.append(target);
+
+    let result = call(proxy, 'check_remote_timestamp', calldata).unwrap();
+    assert(*result.at(0_u32) == 100, *result.at(0_u32));
+}
+
+#[test]
+fn test_start_warp_with_invoke() {
+    let deployed_contract_address = deploy_contract('storing_timestamp', ArrayTrait::new()).unwrap();
+    assert(deployed_contract_address != 0, 'deployed_contract_address != 0');
+
+    let result = call(deployed_contract_address, 'retrieve_stored', ArrayTrait::new()).unwrap();
+    assert(*result.at(0_u32) == 0, *result.at(0_u32));
+    start_warp(100, deployed_contract_address);
+
+    let result = invoke(deployed_contract_address, 'store_timestamp', ArrayTrait::new()).unwrap();
+
+    let result = call(deployed_contract_address, 'retrieve_stored', ArrayTrait::new()).unwrap();
+    assert(*result.at(0_u32) == 100, *result.at(0_u32));
+}
+
+#[test]
+fn test_start_warp_constructor() {
+    let class_hash = declare('storing_constructor_timestamp').unwrap();
+    let prepared = prepare(class_hash, ArrayTrait::new()).unwrap();
+
+    start_warp(100, prepared.contract_address);
+
+    let deployed_contract_address = deploy(prepared).unwrap();
+    let result = call(deployed_contract_address, 'retrieve_stored', ArrayTrait::new()).unwrap();
+    assert(*result.at(0_u32) == 100, *result.at(0_u32));
+}
+
+#[test]
+fn test_start_stop_warp() {
+    let deployed_contract_address = deploy_contract('simple', ArrayTrait::new()).unwrap();
+    assert(deployed_contract_address != 0, 'deployed_contract_address != 0');
+
+    let result = call(deployed_contract_address, 'check_timestamp', ArrayTrait::new()).unwrap();
+    assert(*result.at(0_u32) == 0, *result.at(0_u32));
+
+    start_warp(100, deployed_contract_address);
+    let result = call(deployed_contract_address, 'check_timestamp', ArrayTrait::new()).unwrap();
+    assert(*result.at(0_u32) == 100, *result.at(0_u32));
+
+    stop_warp(deployed_contract_address);
+    let result = call(deployed_contract_address, 'check_timestamp', ArrayTrait::new()).unwrap();
+    assert(*result.at(0_u32) == 0, *result.at(0_u32));
+}
+
+#[test]
+fn test_stop_warp_on_non_existent() {
+    stop_warp(1234).unwrap();
+}
+
+#[test]
+fn test_stop_warp_on_not_warped() {
+    let deployed_contract_address = deploy_contract('simple', ArrayTrait::new()).unwrap();
+
+    stop_warp(1234).unwrap();
+}
+
+#[test]
+fn test_stop_warp_multiple_times() {
+    let deployed_contract_address = deploy_contract('simple', ArrayTrait::new()).unwrap();
+    start_warp(123, deployed_contract_address).unwrap();
+
+    stop_warp(deployed_contract_address).unwrap();
+    stop_warp(deployed_contract_address).unwrap();
+}
```

Comment:
We could also check what happens on multiple `start_warps` for the same contract, preferably with different value per call so we can verify if the last call is used.

Reply:
[Add a test for warp using only last value](https://github.com/software-mansion/protostar/pull/1867/commits/ac80010ba47557c1f1142c91be9e787cb1a9ebe0)